src/OrchardCore/OrchardCore.Indexing.Abstractions/DocumentIndex.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+
+using Microsoft.AspNetCore.Html;
+
+namespace OrchardCore.Indexing
+{
+    public class DocumentIndex
+    {
+        public DocumentIndex(string contentItemId)
+        {
+            ContentItemId = contentItemId;
+        }
+
+        public List<IDocumentIndexEntry> Entries { get; } = new List<IDocumentIndexEntry>();
+
+        public void Set(string name, string value, DocumentIndexOptions options)
+        {
+            Entries.Add(new DocumentIndexEntry(name, value, Types.Text, options));
+        }
+
+        public void Set(string name, IHtmlContent value, DocumentIndexOptions options)
+        {
+            Entries.Add(new DocumentIndexEntry(name, value, Types.Text, options));
+        }
+
+        public void Set(string name, DateTimeOffset? value, DocumentIndexOptions options)
+        {
+            Entries.Add(new DocumentIndexEntry(name, value, Types.DateTime, options));
+        }
+
+        public void Set(string name, int? value, DocumentIndexOptions options)
+        {
+            Entries.Add(new DocumentIndexEntry(name, value, Types.Integer, options));
+        }
+
+        public void Set(string name, bool? value, DocumentIndexOptions options)
+        {
+            Entries.Add(new DocumentIndexEntry(name, value, Types.Boolean, options));
+        }
+
+        public void Set(string name, double? value, DocumentIndexOptions options)
+        {
+            Entries.Add(new DocumentIndexEntry(name, value, Types.Number, options));
+        }
+
+        public void Set(string name, decimal? value, DocumentIndexOptions options)
+        {
+            Entries.Add(new DocumentIndexEntry(name, value, Types.Number, options));
+        }
+
+        public string ContentItemId { get; }
+
+        public enum Types
+        {
+            Integer,
+            Text,
+            DateTime,
+            Boolean,
+            Number
+        }
+
+        public interface IDocumentIndexEntry
+        {
+            string Name { get; }
+            object Value { get; }
+            Types Type { get; }
+            DocumentIndexOptions Options { get; }
+        }
+
+        private class DocumentIndexEntry : IDocumentIndexEntry
```

Comment:
Not sure there is a need to an interface. It's just a data object. We don't even do that anymore for services if there is no need to be able to replace them, or test them.

Reply:
OK good point ... I'll remove the interface and revert the abstraction.

Reply:
Good point. I will revert the interface. That should revert quite a bit of code.